### DISCUSSION TITLE
Sign driver package

### DISCUSSION THREAD

**Discussion by Emma1963** [winfsp/winfsp#584] [General]
Hi @billziss-gh
I would like to sign my rebrand driver, then we can use it in our application. I would like to ask several questions for a clarification.
I referred to this article https://winfsp.dev/doc/WinFsp-Building/ , but there is only one person in our company who can sign with an EV certificate. So I need to change the steps.

1. Are the steps correct?
    -   1. First I build the projects to get unsigned <rebrandname>fsp-x64.sys, <rebrandname>fsp-x86.sys, <rebrandname>fsp-a64.sys 
    -   2. The build also gets driver-a64.inf, driver-x64.inf, driver-x86.inf
    -   3. Get three .cat files for all the. inf files
    -   4. Add the three .sys files, and the three .inf files into the driver.cab with.ddf file    (**Do we need to add the three .cat files, I think the answer is yes**?)
    -   5. Sign with EV certificate for the driver.cab. 
    -   6. Upload the signed driver.cab file to Microsoft for "attestation signing".
    -   7. Download and extract the three .sys files.
    -   8. The .inf files are not used anymore. 
2. Do I need to sign <rebrandname>fsp-a64.dll, <rebrandname>fsp-x64.dll,  <rebrandname>fsp-x86.dll and the three launcher exe as the steps? Do they even need to be signed with the EV certificate?
3. I saw all the dlls under the bin folder with two digital signatures. Can you explain what are they?
![Screenshot 2024-10-22 225905](https://github.com/user-attachments/assets/e00c96d8-b556-40b7-8225-f5e685737134)

**Comment by billziss-gh** [comment]
@Emma1963 please contact me directly for commercial level support. It must be understood that I cannot provide this level of detailed support for free over a public forum.

**Reply by Emma1963** [reply]
Sure, I will contact you directly.
